packages/flutter/lib/src/rendering/viewport.dart

```diff
@@ -505,11 +505,18 @@ abstract class RenderViewportBase<ParentDataClass extends ContainerParentDataMix
   @override
   Rect describeApproximatePaintClip(RenderSliver child) {
     final Rect viewportClip = Offset.zero & size;
-    if (child.constraints.overlap == 0) {
+    // The child's viewportMainAxisExtent can be infinite when a
+    // RenderShrinkWrappingViewport is given infinite constraints.
+    //
+    // For example, a shrink wrapping render sliver may have infinite
```

Comment:
I believe this is not correct. When the viewport has infinite constraints it does *not* scroll. It has infinite space to layout its children so there’s no need to scroll. Since it will not be scrolling there also will not be any overlap. 

At this point the viewport basically turns into a regular column/row (depending on orientation).

Reply:
It's not the viewport that has infinite constraints, it's the sliver (child).  Does that make things better?

Reply:
I thought that the mainAxisExtent can only be infinite if the viewport itself also has infinite constraints and hence is not scrollable.

Reply:
We talked offline and updated the comment a bit.  The real catch here is that the scroll physics can end up making this scroll just a little bit (e.g. bouncing scroll physics trying to move it up/down).